# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-xcb-screen.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-xcb-screen.c
--- a/gfx/cairo/cairo/src/cairo-xcb-screen.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-xcb-screen.c	Mon May 06 19:25:12 2024 +0000
@@ -121,7 +121,7 @@
     cairo_font_options_set_hint_style (&screen->font_options, hint_style);
     cairo_font_options_set_antialias (&screen->font_options, antialias);
     cairo_font_options_set_subpixel_order (&screen->font_options, subpixel_order);
-    cairo_font_options_set_lcd_filter (&screen->font_options, lcd_filter);
+    _cairo_font_options_set_lcd_filter (&screen->font_options, lcd_filter);
     cairo_font_options_set_hint_metrics (&screen->font_options, CAIRO_HINT_METRICS_ON);
 }
 
@@ -169,7 +169,7 @@
 
     for (i = 0; i < ARRAY_LENGTH (screen->gc); i++) {
 	if (screen->gc_depths[i] != 0)
-	    _cairo_xcb_connection_free_gc (screen->connection, screen->gc[i]);
+	    xcb_free_gc (screen->connection->xcb_connection, screen->gc[i]);
     }
 
     _cairo_cache_fini (&screen->linear_pattern_cache);
@@ -350,7 +350,7 @@
     if (i == ARRAY_LENGTH (screen->gc)) {
 	/* perform random substitution to ensure fair caching over depths */
 	i = rand () % ARRAY_LENGTH (screen->gc);
-	_cairo_xcb_connection_free_gc (screen->connection, screen->gc[i]);
+	xcb_free_gc (screen->connection->xcb_connection, screen->gc[i]);
     }
 
     screen->gc[i] = gc;